packages/nx/src/lock-file/yarn.ts

```diff
@@ -288,6 +297,51 @@ function pruneDependencies(
   return result;
 }
 
+function pruneDependency(
+  packageName: string,
+  normalizedPackageJson: PackageJsonDeps,
+  dependencies: LockFileData['dependencies'],
+  result: LockFileData['dependencies'],
+  isPatch = false
```

Comment:
Could `isPatch` be inferred?

```
const isPatch = packageName.includes(`@patch:`)
```

Or

```
const patchMatch = packageName.match(/(.+)@patch:(.+)/);
// maybe `packageName.match(/(?<pkg>.+)@patch:(.+)/)` for named group
if (patchMatch) {
  const originalPackageName = patchMatch[1]; // patchMatch.groups['pkg'] for named group
  //...
} else {
  //...
}
```

Reply:
We could, but for performance reasons, it's easier just to pass the boolean, since we know at call whether it's a patch or not.